esmvalcore/cmor/_fixes/cordex/cordex_fixes.py

```diff
@@ -166,6 +182,151 @@ def _fix_geographical_coords(self, cube, domain):
+            np.array(domain_bounds['lats']))
+
+        x_coord = cube.coord(var_name='x')
+        y_coord = cube.coord(var_name='y')
+        x_coord.convert_units('m')
+        y_coord.convert_units('m')
+
+        xmin, xmax = x_coord.points.min(), x_coord.points.max()
+        ymin, ymax = y_coord.points.min(), y_coord.points.max()
+
+        # Lambdas not allowed.
+        def dist(point_a, point_b):
+            return np.linalg.norm(np.array(point_a) - np.array(point_b))
+
+        # Compare the 4 corners of cube domain vs. the standard domain ones.
+        cube_corners = np.take(lambert_bounds, indices=[0, 2, 6, 8], axis=0)
+        domain_corners = np.array([
+            [xmin, ymax], [xmax, ymax], [xmin, ymin], [xmax, ymin]])
+        return np.mean(
+            [dist(x, y) for x, y in zip(cube_corners, domain_corners)]) < 1e6
+
+    def _check_lambert_conformal_geog_coords(
+            self, cube, domain_bounds):
+        """Check lambert conformal geographical coordinates."""
+        lons = np.array(domain_bounds['lons'])
+        lats = np.array(domain_bounds['lats'])
+        cube2d = cube[0]
+        cube_x_std = extract_region(cube2d,
+                                    start_longitude=lons.min(),
+                                    end_longitude=lons.max(),
+                                    start_latitude=lats.min(),
+                                    end_latitude=lats.max())
+        return (np.sum(cube2d.shape) / np.sum(cube_x_std.shape)) < 1.15
+
+    def _fix_lambert_projection_coord_using_geographical(self, cube):
+        """Use geographical coords to retrieve projection coords."""
+        # Retrieve infos from cube.
+        lambert_system = cube.coord_system().as_cartopy_crs()
+        lon_pts = cube.coord("longitude").points
+        lat_pts = cube.coord("latitude").points
+
+        # lonlat are in geographical and Dim coords are in lambert system.
+        lonlat_lambert = self._transform_points(crs_from=GEOG_SYSTEM,
+                                                crs_to=lambert_system,
+                                                x_data=lon_pts,
+                                                y_data=lat_pts)
+
+        # Derived from the way aux_coords are built.
+        x_points = np.average(lonlat_lambert[:, :, 0], axis=0)
+        y_points = np.average(lonlat_lambert[:, :, 1], axis=1)
+
+        table = CMOR_TABLES["CORDEX"]
+
+        # For each coord, make new and replace.
+        for var_name, points in zip(['x', 'y'], [x_points, y_points]):
+            coord = self._make_projection_coord(points,
+                                                table.coords[var_name],
+                                                cube.coord_system())
+            old_coord = cube.coord(var_name=var_name)
+            dim = old_coord.cube_dims(cube)
+            cube.remove_coord(old_coord)
+            cube.add_dim_coord(coord, dim)
+
+    def _fix_geographical_coord_using_projection(self, cube):
```

Comment:
there is no back and forth:
- projection_x/y_coordinate points and bounds are numpy arrays
- they are turned into two dasks meshgrids
- from this point on, all the computations are lazy

Reply:
Oh you are absolutely right.
I checked and it appears that the pyproj `Transformer.transform` function is materializing the input arrays.
when looking at the source code, the function is accessing the `__array__()` getter of the inputs which in the case of dask arrays is forcing the materialization. 

Since it is unavoidable, the next version of `AllVars._transform_points` will not include casting to dask array, I will move the casting to just before plugging the points in the coords.  

Sorry for my initial uninformed answer.